components/analytics/true_up_review.tsx

```diff
@@ -0,0 +1,217 @@
+        return moment(date).format('MMMM DD, YYYY');
+    };
+
+    const handleSubmitReview = () => {
+        dispatch(getTrueUpReviewBundle());
+        dispatch(getTrueUpReviewStatus());
+        setSubmitted(true);
+    };
+
+    const handleDownloadBundle = () => {
+        dispatch(getTrueUpReviewBundle());
+
+        // Set submitted to true download the bundle data via useEffect
+        // (the selector for the review profile would only return the initial state without doing this).
+        setSubmitted(true);
+    };
+
+    const dueDate = (
+        <div className='dueDate'>
+            <span>
+                <FormattedMessage
+                    id='admin.billing.trueUpReview.due_date'
+                    defaultMessage='Due '
+                />
+            </span>
+            <span>
+                {formattedDueDate()}
+            </span>
+        </div>
+    );
+
+    const submitButton = (
+        <button
+            className={trueUpReviewError ? 'Submit_Button_Error' : 'Submit_Button'}
+            onClick={isAirGapped ? handleDownloadBundle : handleSubmitReview}
+        >
+            {isAirGapped ?
+                <FormattedMessage
+                    id='admin.billing.trueUpReview.button_download'
+                    defaultMessage='Download Data'
+                /> :
+                <FormattedMessage
+                    id='admin.billing.trueUpReview.button_share'
+                    defaultMessage='Share to Mattermost'
+                />
+            }
+        </button>
+    );
+
+    const errorStatus = (
+        <>
+            <WarningIcon additionalClassName={'warning-icon'}/>
+            <FormattedMessage
+                id='admin.billing.trueUpReview.submit_error'
+                defaultMessage='There was an issue sending your True Up Review. Please try again.'
+            />
+            {submitButton}
+        </>
+    );
+
+    const successStatus = (
+        <>
+            <CheckMarkSvg/>
+            <FormattedMessage
+                id='admin.billing.trueUpReview.submit_success'
+                defaultMessage='Success!'
+            />
+            <FormattedMessage
+                id='admin.billing.trueUpReview.submit.thanks_for_sharing'
+                defaultMessage='Thanks for sharing data needed for your true-up review.'
+            />
+        </>
+    );
+
+    const reviewDetails = (
+        <>
+            {dueDate}
+            <FormattedMessage
+                id='admin.billing.trueUpReview.share_data_for_review'
+                defaultMessage='Share the below workspace data with Mattermost for your quarterly true-up Review.'
+            />
+            {submitButton}
+        </>
+    );
+
+    const cardContent = () => {
+        if (trueUpReviewError) {
+            return errorStatus;
+        }
+
+        // If the due date is empty we still have the default state.
+        if (reviewStatus.due_date === 0) {
```

Comment:
Nit: Check also the status of the request. I am curious to know the response if we aren't in the date range for this. I'm more fan of setting the request status in redux, as we have listeners for the request, error, and success. An example of this is this [reducer](https://github.com/mattermost/mattermost-webapp/blob/1607dcddafe53ad2be4836e859c04eb5f0cc2a66/packages/types/src/cloud.ts#L105), and then you can do this [kind of checks](https://github.com/mattermost/mattermost-webapp/blob/1607dcddafe53ad2be4836e859c04eb5f0cc2a66/components/announcement_bar/overage_users_banner/index.tsx#L80)

Reply:
Interesting. I found the error state (and am currently using it) but I didn't realize we have functionality for checking the request status. I'll see if I can get this added in to my current changes.

Reply:
As for the response, it will always contain the next due date and an indicator of wether or not the review is complete.
The front-end is the only controller of what happens if we are not within range of the due date (2 weeks).